x/staking/keeper/test_common.go

```diff
@@ -35,22 +43,26 @@ func TestingUpdateValidator(keeper *Keeper, ctx sdk.Context, validator types.Val
 				deleted = true
 			}
 
-			store.Delete(iterator.Key())
+			if err = store.Delete(iterator.Key()); err != nil {
+				panic(err)
+			}
 		}
 	}
 
-	keeper.SetValidatorByPowerIndex(ctx, validator)
+	if err = keeper.SetValidatorByPowerIndex(ctx, validator); err != nil {
+		panic(err)
+	}
 
 	if !apply {
 		ctx, _ = ctx.CacheContext()
 	}
-	_, err := keeper.ApplyAndReturnValidatorSetUpdates(ctx)
+	_, err = keeper.ApplyAndReturnValidatorSetUpdates(ctx)
 	if err != nil {
 		panic(err)
 	}
 
-	validator, found := keeper.GetValidator(ctx, validator.GetOperator())
-	if !found {
+	validator, err = keeper.GetValidator(ctx, validator.GetOperator())
+	if err != nil {
 		panic("validator expected but not found")
```

Comment:
better error check needed here, or we should use the err instead of that string.

Reply:
panicking with the err now